core/src/test/java/org/apache/iceberg/view/TestViewMetadata.java

```diff
@@ -0,0 +1,174 @@
+
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Arrays;
+import java.util.Collection;
+import org.apache.iceberg.Schema;
+import org.apache.iceberg.relocated.com.google.common.collect.ImmutableList;
+import org.apache.iceberg.relocated.com.google.common.collect.ImmutableMap;
+import org.apache.iceberg.types.Types;
+import org.assertj.core.api.Assertions;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runners.Parameterized;
+
+public class TestViewMetadata {
+
+  private static ViewVersion version1 =
+      ImmutableViewVersion.builder()
+          .versionId(1)
+          .timestampMillis(4353L)
+          .summary(ImmutableMap.of("operation", "create"))
+          .schemaId(1)
+          .addRepresentations(
+              ImmutableSQLViewRepresentation.builder()
+                  .sql("select 'foo' foo")
+                  .dialect("spark-sql")
+                  .defaultCatalog("some-catalog")
+                  .build())
+          .build();
+
+  private static ViewHistoryEntry historyEntry1 =
+      ImmutableViewHistoryEntry.builder().timestampMillis(4353L).versionId(1).build();
+
+  private static ViewVersion version2 =
+      ImmutableViewVersion.builder()
+          .versionId(2)
+          .schemaId(1)
+          .timestampMillis(5555L)
+          .summary(ImmutableMap.of("operation", "replace"))
+          .addRepresentations(
+              ImmutableSQLViewRepresentation.builder()
+                  .sql("select 1 id, 'abc' data")
+                  .defaultCatalog("some-catalog")
+                  .dialect("spark-sql")
+                  .build())
+          .build();
+
+  private static ViewHistoryEntry historyEntry2 =
+      ImmutableViewHistoryEntry.builder().timestampMillis(5555L).versionId(2).build();
+
+  private static final Schema TEST_SCHEMA =
+      new Schema(
+          1,
+          Types.NestedField.required(1, "x", Types.LongType.get()),
+          Types.NestedField.required(2, "y", Types.LongType.get(), "comment"),
+          Types.NestedField.required(3, "z", Types.LongType.get()));
+
+  @Parameterized.Parameters
+  public static Collection<Object[]> parameters() {
+    return Arrays.asList(new Object[][] {});
+  }
+
+  @Test
+  public void testReadAndWriteValidViewMetadata() throws Exception {
+    String json = readViewMetadataInputFile("view/ValidViewMetadata.json");
+    ViewMetadata expectedViewMetadata =
+        ImmutableViewMetadata.builder()
+            .currentSchemaId(1)
+            .schemas(ImmutableList.of(TEST_SCHEMA))
+            .versions(ImmutableList.of(version1, version2))
+            .history(ImmutableList.of(historyEntry1, historyEntry2))
+            .location("s3://bucket/test/location")
+            .properties(ImmutableMap.of("some-key", "some-value"))
+            .currentVersionId(2)
+            .formatVersion(1)
+            .build();
+    assertSameViewMetadata(expectedViewMetadata, ViewMetadataParser.fromJson(json));
```

Comment:
nit: `assertSame` usually suggests that there's some sort of identity comparison between two objects.